ch3/recursion.js

```diff
@@ -0,0 +1,32 @@
+const assert =  require('assert');
+/**
+ * checks if input value is even
+ * @param {*} n - input value
+ * @returns true if even, false if not
+ */
+const isEven = (n) => {
+  n = Math.abs(n);
+
+  //these are so short. Can i just keep them one line? Would anyone really care?
+  if (n === 0) return true; 
+  else if (n === 1) return false;
+  return isEven(n-2);
+  //which one? first one easier to read
+  /*return (n === 0) ? true :
+  (n === 1) ? false :
+  isEven(n - 2);*/
```

Comment:
always bias towards the easier one to read! the easier it is to read, the easier it is for anyone to maintain. there are some scenarios where making code _fast_ means very unreadable code, but don't worry about those scenarios right now.

Reply:
Ok